PXB-3215 Avoid the use of encryption keys or passwords in docs (8.0) (#217)
percona/pxb-docs: docs/encrypt-backups.md

# Encrypt backups
Percona XtraBackup supports encrypting and decrypting local and streaming backups with the upstream option, adding another protection layer. The
encryption is implemented using the `libgcrypt` library from GnuPG.

## Create encrypted backups

The following options create encrypted backups. The
`--encrypt-key` and `--encrypt-key-file` options specify the encryption key and are mutually exclusive. You should select one or the other.

* `--encrypt`

* `--encrypt-key`

* `--encrypt-key-file``

You can use a command, for example,

> `openssl rand -base64 24`, to generate a randomly generated alphanumeric string as an encryption key.

Reviewer: ```suggestion
`openssl rand -base64 24` to generate a random alphanumeric string as an encryption key.
```

### The `--encrypt-key` option

An example of the *xtrabackup* command using the `--encrypt-key`:

```{.bash data-prompt="$"}
$  xtrabackup --backup --encrypt=AES256 --encrypt-key="{randomly-generated-alphanumeric-string}" --target-dir=/data/backup
```

### The `--encrypt-key-file` option

> Be aware that your text editor can automatically insert an CRLF (end of line) character in the `KEYFILE`. This character makes the key invalid because the size is wrong. The recommended method uses the command line: `echo -n “{randomly-generated-alphanumeric-string}” > /data/backups/keyfile` to create the file.

Reviewer: ```suggestion
Be aware that your text editor can automatically insert a CRLF (end of line) character in the `KEYFILE`. This character makes the key invalid because the length is wrong. The recommended method uses the command line: `echo -n “{randomly-generated-alphanumeric-string}” > /data/backups/keyfile` to create the file.
```

An example of using the `--encrypt-key-file` option:

```{.bash data-prompt="$"}
$ xtrabackup --backup --encrypt=AES256 --encrypt-key-file=/data/backups/keyfile --target-dir=/data/backup
```

## Optimize the encryption process

Two encrypted backup options, `--encrypt-threads` and
`--encrypt-chunk-size`, can speed up
the encryption process. 

> Using the `--encrypt-threads` option specifies that multiple threads can be used for encryption in parallel. 

Reviewer: ```suggestion
Use the `--encrypt-threads` option to enable parallel encryption with multiple threads.
```

The `--encrypt-chunk-size` option specifies the size, in bytes, of the working encryption buffer for each encryption thread. The default size is 64K.

## Decrypt encrypted backups

> You can decrypt backups the `xbcrypt` binary. The following example encrypts a backup.

Reviewer: ```suggestion
You can decrypt backups with the `xbcrypt` binary. The following example encrypts a backup.
```
